api/src/test/java/org/apache/iceberg/expressions/TestPredicateBinding.java

```diff
@@ -134,7 +134,7 @@ public void testLiteralConversion() {
       assertThat(bound.isLiteralPredicate()).isTrue();
       assertThat(bound.asLiteralPredicate().literal().value())
           .as("Should convert literal value to decimal")
-          .isEqualTo(new BigDecimal("12.40"));
+          .isEqualTo("12.40");
```

Comment:
we can leave the BigDecimal stuff because `bound.asLiteralPredicate().literal().value()` will produce a `BigDecimal` and so we're expecting a `BigDecimal` here. The only reason this test passes is actually because `isEqualTo()` will convert that String to a `BigDecimal` again

Reply:
Thank you for the explanation @nastra